Indicators/AwesomeOscillator.cs

```diff
@@ -0,0 +1,101 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data.Market;
+using System;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// The Awesome Oscillator Indicator tracks the price midpoint-movement of a security. Specifically,
+    /// <para>
+    /// AO = MAfast[(H+L)/2] - MAslow[(H+L)/2] 
+    /// </para>
+    /// where MAfast and MAslow denote simple moving averages wherein fast has a shorter period.
+    /// https://www.barchart.com/education/technical-indicators/awesome_oscillator
+    /// </summary>
+    public class AwesomeOscillator : BarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+        /// <summary>
+        /// Gets the indicators slow period moving average.
+        /// </summary>
+        private IndicatorBase<IndicatorDataPoint> SlowAo { get; }
+
+        /// <summary>
+        /// Gets the indicators fast period moving average.
+        /// </summary>
+        private IndicatorBase<IndicatorDataPoint> FastAo { get; }
```

Comment:
If these members are private, they should not be properties, but `readonly` fields:
```csharp
private readonly IndicatorBase<IndicatorDataPoint> _slowAo;
private readonly IndicatorBase<IndicatorDataPoint> _fastAo;
```
Or should they be (readonly) public properties?
```csharp
public IndicatorBase<IndicatorDataPoint> SlowAo { get; }
public IndicatorBase<IndicatorDataPoint> FastAo { get; }
```

Reply:
I did `private IndicatorBase<IndicatorDataPoint> FastAo { get; } ` mostly for lazy loading (until data is first added). Regardless, I think making them public is an even better idea for those wishing to plot the two separately.